backend/src/crd/kubernetes/v2beta1/pipeline_types.go

```diff
@@ -0,0 +1,102 @@
+/*
+Copyright 2025.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v2beta1
+
+import (
+	"github.com/kubeflow/pipelines/backend/src/apiserver/model"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+)
+
+// PipelineSpec defines the desired state of Pipeline.
+type PipelineSpec struct {
+	Description string `json:"description,omitempty"`
+}
+
+type PipelineStatus struct {
+	Conditions []metav1.Condition `json:"conditions,omitempty"`
+}
+
+// +kubebuilder:object:root=true
+// +kubebuilder:subresource:status
+
+// Pipeline is the Schema for the pipelines API.
+type Pipeline struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	Spec   PipelineSpec   `json:"spec,omitempty"`
+	Status PipelineStatus `json:"status,omitempty"`
```

Comment:
@rimolive Could you explain why is the PipelineStatus needed here and how it's being used?  Just trying to understand its purpose better.

Reply:
Good catch! I thought I removed that type...

Reply:
Agreed. I think it can also be removed.